File: backend/ngts/src/main/java/com/three/ngts/Entity/Name.java
package com.three.ngts.Entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "name")
public class Name {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "name_id")
    private Long nameId;

    @Column(name = "first_name")
    private String firstName;

    @Column(name = "last_name")
    private String lastName;

    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;
}

File: frontend/next-generation-ticketing-system/src/Entity/Name.ts
import User from "./User";

class Name {
  nameId: number;
  firstName: string;
  lastName: string;
  user: User;

  constructor(nameId: number, firstName: string, lastName: string, user: User) {
    this.nameId = nameId;
    this.firstName = firstName;
    this.lastName = lastName;
    this.user = user;
  }
}

export default Name;
